src/object/cli_obj.c

```diff
@@ -1618,8 +1623,18 @@ obj_req_get_tgts(struct dc_object *obj, enum obj_rpc_opc opc, int *shard,
 
 				rc *= obj->cob_grp_size;
 				if (*shard < rc ||
-				    *shard >= rc + obj->cob_grp_size)
+				    *shard >= rc + obj->cob_grp_size) {
+					D_ERROR("Fetch from invalid shard, "
+						"grp size %u, shard cnt %u, "
+						"grp idx %u, given shard %u, "
+						"dkey hash %lu, dkey %s\n",
+						obj->cob_grp_size,
+						obj->cob_shards_nr,
+						rc / obj->cob_grp_size, *shard,
+						dkey_hash,
+						(char *)dkey->iov_buf);
```

Comment:
some dkey is not printable, and some cases may pass NULL dkey(like in dc_obj_sync)?

Reply:
It is only for DAOS_OBJ_RPC_FETCH case.